### ISSUE TITLE
Make work with Java 9

### ISSUE THREAD

**Issue by tibbe** [openrocket/openrocket#347]
This is an umbrella issue for making OpenRocket work with Java 9, once it's released. Supporting Java 9 has several benefits, perhaps foremost making OpenRocket work on HiDPI screens.

There are several issues to resolve e.g. #346. The major ones relates to how we build the final jar (with bundled deps) and how we make sure the right modules (e.g. `java.scripting`, which contains `javax.script.ScriptEngineFactory`) are exposed to the loaded code (since we use a class loader).

**Comment by teyrana** [comment]
haha, this should be relevant: 
https://docs.oracle.com/javase/9/migrate/toc.htm#JSMIG-GUID-D867DCCC-CEB5-4AFA-9D11-9C62B7A3FAB1

> The changes in JDK 9 may impact code that creates class loaders with null (that is, the bootstrap class loader) as the parent class loader and assumes that all platform classes are visible to the parent. Such code may need to be changed to use the platform class loader as the parent (see ClassLoader.getPlatformClassLoader).

**Comment by teyrana** [comment]
Probably solution: 

[ Multi-Release Jars ]( http://openjdk.java.net/jeps/238 )

**Comment by tibbe** [comment]
This is what the failure looks like:

```
java --version
java 10.0.1 2018-04-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.1+10)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.1+10, mixed mode)
```

```
---------- Bug report ----------

Please include a description about what actions you were performing when the exception occurred:




Include your email address (optional; it helps if we can contact you in case we need additional information):




(Do not modify anything below this line.)
---------- Exception stack trace ----------
java.lang.RuntimeException: Error starting OpenRocket
	at net.sf.openrocket.startup.jij.JarInJarStarter.runMain(JarInJarStarter.java:40)
	at net.sf.openrocket.startup.OpenRocket.main(OpenRocket.java:32)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at net.sf.openrocket.startup.jij.JarInJarStarter.runMain(JarInJarStarter.java:38)
	... 1 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.desktop/java.awt.EventQueue.invokeAndWait(Unknown Source)
	at java.desktop/java.awt.EventQueue.invokeAndWait(Unknown Source)
	at java.desktop/javax.swing.SwingUtilities.invokeAndWait(Unknown Source)
	at net.sf.openrocket.startup.SwingStartup.main(SwingStartup.java:76)
	... 6 more
Caused by: com.google.inject.internal.util.$ComputationException: com.google.inject.internal.util.$ComputationException: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:553)
	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:419)
	at com.google.inject.internal.util.$CustomConcurrentHashMap$ComputingImpl.get(CustomConcurrentHashMap.java:2041)
	at com.google.inject.internal.FailableCache.get(FailableCache.java:50)
	at com.google.inject.internal.ConstructorInjectorStore.get(ConstructorInjectorStore.java:49)
	at com.google.inject.internal.ConstructorBindingImpl.initialize(ConstructorBindingImpl.java:125)
	at com.google.inject.internal.InjectorImpl.initializeJitBinding(InjectorImpl.java:521)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(InjectorImpl.java:847)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive(InjectorImpl.java:772)
	at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(InjectorImpl.java:256)
	at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImpl.java:205)
	at com.google.inject.internal.InjectorImpl.getInternalFactory(InjectorImpl.java:853)
	at com.google.inject.internal.FactoryProxy.notify(FactoryProxy.java:46)
	at com.google.inject.internal.ProcessedBindingData.runCreationListeners(ProcessedBindingData.java:50)
	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:133)
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:106)
	at com.google.inject.Guice.createInjector(Guice.java:95)
	at com.google.inject.Guice.createInjector(Guice.java:72)
	at com.google.inject.Guice.createInjector(Guice.java:62)
	at net.sf.openrocket.startup.SwingStartup.runInEDT(SwingStartup.java:143)
	at net.sf.openrocket.startup.SwingStartup.access$000(SwingStartup.java:43)
	at net.sf.openrocket.startup.SwingStartup$1.run(SwingStartup.java:79)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.desktop/java.awt.EventQueue.access$600(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
Caused by: com.google.inject.internal.util.$ComputationException: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:553)
	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:419)
	at com.google.inject.internal.util.$CustomConcurrentHashMap$ComputingImpl.get(CustomConcurrentHashMap.java:2041)
	at com.google.inject.internal.FailableCache.get(FailableCache.java:50)
	at com.google.inject.internal.MembersInjectorStore.get(MembersInjectorStore.java:65)
	at com.google.inject.internal.ConstructorInjectorStore.createConstructor(ConstructorInjectorStore.java:73)
	at com.google.inject.internal.ConstructorInjectorStore.access$000(ConstructorInjectorStore.java:28)
	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:36)
	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:32)
	at com.google.inject.internal.FailableCache$1.apply(FailableCache.java:39)
	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:549)
	... 35 more
Caused by: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Unknown Source)
	at java.base/java.lang.Class.getDeclaredMethods(Unknown Source)
	at com.google.inject.spi.InjectionPoint.getInjectionPoints(InjectionPoint.java:662)
	at com.google.inject.spi.InjectionPoint.forInstanceMethodsAndFields(InjectionPoint.java:356)
	at com.google.inject.internal.ConstructorBindingImpl.getInternalDependencies(ConstructorBindingImpl.java:151)
	at com.google.inject.internal.InjectorImpl.getInternalDependencies(InjectorImpl.java:585)
	at com.google.inject.internal.InjectorImpl.cleanup(InjectorImpl.java:543)
	at com.google.inject.internal.InjectorImpl.initializeJitBinding(InjectorImpl.java:529)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(InjectorImpl.java:847)
	at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive(InjectorImpl.java:772)
	at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(InjectorImpl.java:256)
	at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImpl.java:205)
	at com.google.inject.internal.InjectorImpl.getInternalFactory(InjectorImpl.java:853)
	at com.google.inject.internal.SingleFieldInjector.<init>(SingleFieldInjector.java:41)
	at com.google.inject.internal.MembersInjectorStore.getInjectors(MembersInjectorStore.java:125)
	at com.google.inject.internal.MembersInjectorStore.createWithListeners(MembersInjectorStore.java:95)
	at com.google.inject.internal.MembersInjectorStore.access$000(MembersInjectorStore.java:34)
	at com.google.inject.internal.MembersInjectorStore$1.create(MembersInjectorStore.java:42)
	at com.google.inject.internal.MembersInjectorStore$1.create(MembersInjectorStore.java:39)
	at com.google.inject.internal.FailableCache$1.apply(FailableCache.java:39)
	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:549)
	... 45 more
Caused by: java.lang.ClassNotFoundException: javax.script.ScriptEngineFactory
	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 67 more

---------- Thread information ----------
Thread[main,5,main]

---------- System information ----------
OpenRocket version: 15.03
OpenRocket source: default
OpenRocket location: C:\Users\tibbe\Downloads\OpenRocket-15.03.jar
JOGL version: 2.1.5
Current default locale: en_US
System properties:
  awt.toolkit=sun.awt.windows.WToolkit
  file.encoding=Cp1252
  file.encoding.pkg=sun.io
  file.separator=\
  java.awt.graphicsenv=sun.awt.Win32GraphicsEnvironment
  java.awt.printerjob=sun.awt.windows.WPrinterJob
  java.class.path=OpenRocket-15.03.jar
  java.class.version=54.0
  java.home=C:\Program Files\Java\jre-10.0.1
  java.io.tmpdir=C:\Users\tibbe\AppData\Local\Temp\
  java.library.path=C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\Sun\Java\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Intel\iCLS Client\;C:\ProgramData\Oracle\Java\javapath;C:\Program Files\Intel\iCLS Client\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\Program Files\dotnet\;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\IPT;C:\Program Files\Intel\Intel(R) Management Engine Components\IPT;C:\WINDOWS\System32\OpenSSH\;C:\Users\tibbe\AppData\Local\Microsoft\WindowsApps;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\LOVE;;.
  java.runtime.name=Java(TM) SE Runtime Environment
  java.runtime.version=10.0.1+10
  java.specification.name=Java Platform API Specification
  java.specification.vendor=Oracle Corporation
  java.specification.version=10
  java.vendor=Oracle Corporation
  java.vendor.url=http://java.oracle.com/
  java.vendor.url.bug=http://bugreport.java.com/bugreport/
  java.vendor.version=18.3
  java.version=10.0.1
  java.version.date=2018-04-17
  java.vm.compressedOopsMode=Zero based
  java.vm.info=mixed mode
  java.vm.name=Java HotSpot(TM) 64-Bit Server VM
  java.vm.specification.name=Java Virtual Machine Specification
  java.vm.specification.vendor=Oracle Corporation
  java.vm.specification.version=10
  java.vm.vendor="Oracle Corporation"
  java.vm.version=10.0.1+10
  jdk.debug=release
  line.separator=\u000d\u000a
  os.arch=amd64
  os.name=Windows 10
  os.version=10.0
  path.separator=;
  sun.arch.data.model=64
  sun.awt.enableExtraMouseButtons=true
  sun.awt.exception.handler=net.sf.openrocket.gui.main.SwingExceptionHandler$AwtHandler
  sun.boot.library.path=C:\Program Files\Java\jre-10.0.1\bin
  sun.cpu.endian=little
  sun.cpu.isalist=amd64
  sun.desktop=windows
  sun.io.unicode.encoding=UnicodeLittle
  sun.java.command=OpenRocket-15.03.jar
  sun.java.launcher=SUN_STANDARD
  sun.jnu.encoding=Cp1252
  sun.management.compiler=HotSpot 64-Bit Tiered Compilers
  sun.os.patch.level=
  sun.stderr.encoding=cp437
  sun.stdout.encoding=cp437
  user.country=US
  user.dir=C:\Users\tibbe\Downloads
  user.home=C:\Users\tibbe
  user.language=en
  user.name=tibbe
  user.script=
  user.timezone=Europe/Berlin
  user.variant=
---------- Error log ----------
   1      0.001 INFO   (SwingStartup.java:61) Starting up OpenRocket version 15.03
   2      0.001 INFO   (SwingStartup.java:64) Checking for graphics head
   3      0.004 INFO   (SwingStartup.java:75) Moving startup to EDT
   4      0.175 INFO   (SwingStartup.java:129) Initializing the splash screen
   5      0.179 INFO   (SwingStartup.java:133) Registering exception handler
   6      0.182 INFO   (SwingStartup.java:139) Loading databases
   7      0.889 ERROR  (SwingExceptionHandler.java:48) Handling uncaught exception on thread=Thread[main,5,main]
   > java.lang.RuntimeException: Error starting OpenRocket
   > 	at net.sf.openrocket.startup.jij.JarInJarStarter.runMain(JarInJarStarter.java:40)
   > 	at net.sf.openrocket.startup.OpenRocket.main(OpenRocket.java:32)
   > Caused by: java.lang.reflect.InvocationTargetException
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
   > 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
   > 	at net.sf.openrocket.startup.jij.JarInJarStarter.runMain(JarInJarStarter.java:38)
   > 	... 1 more
   > Caused by: java.lang.reflect.InvocationTargetException
   > 	at java.desktop/java.awt.EventQueue.invokeAndWait(Unknown Source)
   > 	at java.desktop/java.awt.EventQueue.invokeAndWait(Unknown Source)
   > 	at java.desktop/javax.swing.SwingUtilities.invokeAndWait(Unknown Source)
   > 	at net.sf.openrocket.startup.SwingStartup.main(SwingStartup.java:76)
   > 	... 6 more
   > Caused by: com.google.inject.internal.util.$ComputationException: com.google.inject.internal.util.$ComputationException: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
   > 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:553)
   > 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:419)
   > 	at com.google.inject.internal.util.$CustomConcurrentHashMap$ComputingImpl.get(CustomConcurrentHashMap.java:2041)
   > 	at com.google.inject.internal.FailableCache.get(FailableCache.java:50)
   > 	at com.google.inject.internal.ConstructorInjectorStore.get(ConstructorInjectorStore.java:49)
   > 	at com.google.inject.internal.ConstructorBindingImpl.initialize(ConstructorBindingImpl.java:125)
   > 	at com.google.inject.internal.InjectorImpl.initializeJitBinding(InjectorImpl.java:521)
   > 	at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(InjectorImpl.java:847)
   > 	at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive(InjectorImpl.java:772)
   > 	at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(InjectorImpl.java:256)
   > 	at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImpl.java:205)
   > 	at com.google.inject.internal.InjectorImpl.getInternalFactory(InjectorImpl.java:853)
   > 	at com.google.inject.internal.FactoryProxy.notify(FactoryProxy.java:46)
   > 	at com.google.inject.internal.ProcessedBindingData.runCreationListeners(ProcessedBindingData.java:50)
   > 	at com.google.inject.internal.InternalInjectorCreator.initializeStatically(InternalInjectorCreator.java:133)
   > 	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:106)
   > 	at com.google.inject.Guice.createInjector(Guice.java:95)
   > 	at com.google.inject.Guice.createInjector(Guice.java:72)
   > 	at com.google.inject.Guice.createInjector(Guice.java:62)
   > 	at net.sf.openrocket.startup.SwingStartup.runInEDT(SwingStartup.java:143)
   > 	at net.sf.openrocket.startup.SwingStartup.access$000(SwingStartup.java:43)
   > 	at net.sf.openrocket.startup.SwingStartup$1.run(SwingStartup.java:79)
   > 	at java.desktop/java.awt.event.InvocationEvent.dispatch(Unknown Source)
   > 	at java.desktop/java.awt.EventQueue.dispatchEventImpl(Unknown Source)
   > 	at java.desktop/java.awt.EventQueue.access$600(Unknown Source)
   > 	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
   > 	at java.desktop/java.awt.EventQueue$4.run(Unknown Source)
   > 	at java.base/java.security.AccessController.doPrivileged(Native Method)
   > 	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
   > 	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
   > 	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
   > 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
   > 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
   > 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
   > 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
   > 	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
   > Caused by: com.google.inject.internal.util.$ComputationException: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
   > 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:553)
   > 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:419)
   > 	at com.google.inject.internal.util.$CustomConcurrentHashMap$ComputingImpl.get(CustomConcurrentHashMap.java:2041)
   > 	at com.google.inject.internal.FailableCache.get(FailableCache.java:50)
   > 	at com.google.inject.internal.MembersInjectorStore.get(MembersInjectorStore.java:65)
   > 	at com.google.inject.internal.ConstructorInjectorStore.createConstructor(ConstructorInjectorStore.java:73)
   > 	at com.google.inject.internal.ConstructorInjectorStore.access$000(ConstructorInjectorStore.java:28)
   > 	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:36)
   > 	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:32)
   > 	at com.google.inject.internal.FailableCache$1.apply(FailableCache.java:39)
   > 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:549)
   > 	... 35 more
   > Caused by: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
   > 	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
   > 	at java.base/java.lang.Class.privateGetDeclaredMethods(Unknown Source)
   > 	at java.base/java.lang.Class.getDeclaredMethods(Unknown Source)
   > 	at com.google.inject.spi.InjectionPoint.getInjectionPoints(InjectionPoint.java:662)
   > 	at com.google.inject.spi.InjectionPoint.forInstanceMethodsAndFields(InjectionPoint.java:356)
   > 	at com.google.inject.internal.ConstructorBindingImpl.getInternalDependencies(ConstructorBindingImpl.java:151)
   > 	at com.google.inject.internal.InjectorImpl.getInternalDependencies(InjectorImpl.java:585)
   > 	at com.google.inject.internal.InjectorImpl.cleanup(InjectorImpl.java:543)
   > 	at com.google.inject.internal.InjectorImpl.initializeJitBinding(InjectorImpl.java:529)
   > 	at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(InjectorImpl.java:847)
   > 	at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive(InjectorImpl.java:772)
   > 	at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(InjectorImpl.java:256)
   > 	at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImpl.java:205)
   > 	at com.google.inject.internal.InjectorImpl.getInternalFactory(InjectorImpl.java:853)
   > 	at com.google.inject.internal.SingleFieldInjector.<init>(SingleFieldInjector.java:41)
   > 	at com.google.inject.internal.MembersInjectorStore.getInjectors(MembersInjectorStore.java:125)
   > 	at com.google.inject.internal.MembersInjectorStore.createWithListeners(MembersInjectorStore.java:95)
   > 	at com.google.inject.internal.MembersInjectorStore.access$000(MembersInjectorStore.java:34)
   > 	at com.google.inject.internal.MembersInjectorStore$1.create(MembersInjectorStore.java:42)
   > 	at com.google.inject.internal.MembersInjectorStore$1.create(MembersInjectorStore.java:39)
   > 	at com.google.inject.internal.FailableCache$1.apply(FailableCache.java:39)
   > 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:549)
   > 	... 45 more
   > Caused by: java.lang.ClassNotFoundException: javax.script.ScriptEngineFactory
   > 	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
   > 	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
   > 	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
   > 	... 67 more
  53      0.000 STDERR (-) ===== 45 STDERR lines removed =====
  53      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
  54      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.desktop/java.awt.EventQueue.dispatchEvent(Unknown Source)
  55      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
  56      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
  57      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
  58      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
  59      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.desktop/java.awt.EventDispatchThread.pumpEvents(Unknown Source)
  60      0.893 STDERR (SwingExceptionHandler.java:49) 	at java.desktop/java.awt.EventDispatchThread.run(Unknown Source)
  61      0.893 STDERR (SwingExceptionHandler.java:49) Caused by: com.google.inject.internal.util.$ComputationException: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
  62      0.893 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:553)
  63      0.893 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:419)
  64      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.util.$CustomConcurrentHashMap$ComputingImpl.get(CustomConcurrentHashMap.java:2041)
  65      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.FailableCache.get(FailableCache.java:50)
  66      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.MembersInjectorStore.get(MembersInjectorStore.java:65)
  67      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.ConstructorInjectorStore.createConstructor(ConstructorInjectorStore.java:73)
  68      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.ConstructorInjectorStore.access$000(ConstructorInjectorStore.java:28)
  69      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:36)
  70      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.ConstructorInjectorStore$1.create(ConstructorInjectorStore.java:32)
  71      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.FailableCache$1.apply(FailableCache.java:39)
  72      0.894 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:549)
  73      0.894 STDERR (SwingExceptionHandler.java:49) 	... 35 more
  74      0.894 STDERR (SwingExceptionHandler.java:49) Caused by: java.lang.NoClassDefFoundError: javax/script/ScriptEngineFactory
  75      0.894 STDERR (SwingExceptionHandler.java:49) 	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
  76      0.895 STDERR (SwingExceptionHandler.java:49) 	at java.base/java.lang.Class.privateGetDeclaredMethods(Unknown Source)
  77      0.895 STDERR (SwingExceptionHandler.java:49) 	at java.base/java.lang.Class.getDeclaredMethods(Unknown Source)
  78      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.spi.InjectionPoint.getInjectionPoints(InjectionPoint.java:662)
  79      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.spi.InjectionPoint.forInstanceMethodsAndFields(InjectionPoint.java:356)
  80      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.ConstructorBindingImpl.getInternalDependencies(ConstructorBindingImpl.java:151)
  81      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.getInternalDependencies(InjectorImpl.java:585)
  82      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.cleanup(InjectorImpl.java:543)
  83      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.initializeJitBinding(InjectorImpl.java:529)
  84      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.createJustInTimeBinding(InjectorImpl.java:847)
  85      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.createJustInTimeBindingRecursive(InjectorImpl.java:772)
  86      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.getJustInTimeBinding(InjectorImpl.java:256)
  87      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.getBindingOrThrow(InjectorImpl.java:205)
  88      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.InjectorImpl.getInternalFactory(InjectorImpl.java:853)
  89      0.895 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.SingleFieldInjector.<init>(SingleFieldInjector.java:41)
  90      0.896 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.MembersInjectorStore.getInjectors(MembersInjectorStore.java:125)
  91      0.896 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.MembersInjectorStore.createWithListeners(MembersInjectorStore.java:95)
  92      0.896 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.MembersInjectorStore.access$000(MembersInjectorStore.java:34)
  93      0.896 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.MembersInjectorStore$1.create(MembersInjectorStore.java:42)
  94      0.896 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.MembersInjectorStore$1.create(MembersInjectorStore.java:39)
  95      0.896 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.FailableCache$1.apply(FailableCache.java:39)
  96      0.896 STDERR (SwingExceptionHandler.java:49) 	at com.google.inject.internal.util.$MapMaker$StrategyImpl.compute(MapMaker.java:549)
  97      0.896 STDERR (SwingExceptionHandler.java:49) 	... 45 more
  98      0.896 STDERR (SwingExceptionHandler.java:49) Caused by: java.lang.ClassNotFoundException: javax.script.ScriptEngineFactory
  99      0.896 STDERR (SwingExceptionHandler.java:49) 	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
 100      0.896 STDERR (SwingExceptionHandler.java:49) 	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
 101      0.896 STDERR (SwingExceptionHandler.java:49) 	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
 102      0.896 STDERR (SwingExceptionHandler.java:49) 	... 67 more
 103      0.896 INFO   (SwingExceptionHandler.java:64) Exception handler not on EDT, invoking dialog on EDT
 104      0.898 INFO   (SwingExceptionHandler.java:201) Showing Exception dialog
 105      5.176 USER   (SwingExceptionHandler.java:220) User requested sending bug report
 106      5.179 DEBUG  (JarUtil.java:40) Found jar file using codeSource
---------- End of bug report ----------
```

**Comment by kernelsmith** [comment]
I'm also erroring out while trying to start 15.03 on Linux (Open JDK 11.0.3) and MacOS (Oracle Java 11.0.2). I'm of course assuming this is because Open Rocket only supports 8, a full 3 major versions behind. If we're unlikely to be current-version-compatible anytime soon, we should at least mention the compatibility on the main page and certainly on the packages page http://openrocket.info/packages.html This is my second time trying to get OpenRocket to run and I wasted a good 30 mins trying to track down the stack trace, upgrading packages etc. I really could have benefitted from the simple knowledge about compatibility.

**Comment by lectroidmarc** [comment]
Yeah, I just ran into this on Ubuntu 19.04.  I used to have java-8 installed under 18.10, and was able to run OpenRocket, but it appears there's no java-8 for Ubuntu 19.04 anymore.

**Comment by neilbalch** [comment]
I too need this software to run on >Java8. 😦 If there's any development to be done in order to support higher Java versions, I'd be happy to volunteer my time.

**Comment by tarnh** [comment]
@neilbalch Just do it ;)

**Comment by teyrana** [comment]
Ongoing work at #556

**Comment by teyrana** [comment]
Probably fixed by #572.  (Fix is now in mainline development)

**Comment by itshanney** [comment]
@teyrana propose to close this as mainline development is now on Java 9+.
